File: nysol_python/mcmd/methods/convTex/db/smp/mcut.py
# ================================================
# コマンドマニュアル自動作成用サンプルコードデータ
# ================================================
db={}
db['name']='mcut'

############################### IDATA
db['idatas']=[]

data={}
data['name']='dat1.csv'
data['text']='''
customer,quantity,amount
A,1,10
A,2,20
B,1,15
B,3,10
B,1,20
'''
db['idatas'].append(data)

data={}
data['name']='dat2.csv'
data['text']='''
A,1,10
A,2,20
B,1,15
B,3,10
B,1,20
'''
db['idatas'].append(data)

############################### SCRIPTS
db['scripts']=[]

script={}
script['title']='基本例'
script['comment']='''
``customer`` と ``amount`` 項目を選択する。ただし、 ``amount`` 項目は ``sales`` と名前を変更して出力している。
'''
script['sh_code']='''
mcut f=顧客,金額:売上 i=dat1.csv o=rsl1.csv
'''
script['py_code']='''
nm.mcut(f="customer,amount:sales", i="dat1.csv", o="rsl1.csv").run()
'''
script['odatas']='rsl1.csv'
db['scripts'].append(script)

script={}
script['title']='項目削除'
script['comment']='''
``r=True`` を指定することで、項目を削除できる。
'''
script['sh_code']='''
mcut f=顧客,金額 -r i=dat1.csv o=rsl2.csv
'''
script['py_code']='''
nm.mcut(f="customer,amount", r=True, i="dat1.csv", o="rsl2.csv").run()
'''
script['odatas']='rsl2.csv'
db['scripts'].append(script)

script={}
script['title']='項目名なしデータ'
script['comment']='''
ヘッダなし入力ファイルから、0,2番目の項目を選択し、
``customer`` と ``amount`` という名前で出力する。
'''
script['sh_code']='''
mcut f=0:顧客,2:金額 -nfni i=dat2.csv o=rsl3.csv
'''
script['py_code']='''
nm.mcut(f="0:customer,2:amount", nfni=True, i="dat2.csv", o="rsl3.csv").run()
'''
script['odatas']='rsl3.csv'
db['scripts'].append(script)

File: olddoc/mcmd/en/examples/mcut.rb
#!/usr/bin/env ruby
# coding: utf-8
require "./mkTex.rb"

File.open("dat1.csv","w"){|fpw| fpw.write(
<<'EOF'
customer,quantity,amount
A,1,10
A,2,20
B,1,15
B,3,10
B,1,20
EOF
)}

############## Example 1
title="Basic Example"
comment=<<'EOF'
Extract customer and amount information from the data file \verb|dat1.csv|
Rename the column "amount " to "sales" in the output.
EOF
scp=<<'EOF'
more dat1.csv
mcut f=customer,amount:sales i=dat1.csv o=rsl1.csv
more rsl1.csv
EOF
run(scp,title,comment)

############## Example 2
title="Remove columns"
comment=<<'EOF'
Remove columns customer and amount specified at \verb|-r|.
EOF
scp=<<'EOF'
mcut f=customer,amount -r i=dat1.csv o=rsl2.csv
more rsl2.csv
EOF
run(scp,title,comment)

############## Example 3
title="Data without field names"
comment=<<'EOF'
Select columns 0, 2 from an input file without field header, add \verb|customer| and \verb|amount| as field names in the output file.
EOF
scp=<<'EOF'
mcut f=0:customer,2:amount -nfni i=dat1.csv o=rsl3.csv
more rsl3.csv
EOF
run(scp,title,comment)

